frontend/test/acceptance/pages/EnterDetails.scala

```diff
@@ -7,37 +7,41 @@ import org.scalatest.selenium.Page
 case class EnterDetails(val testUser: TestUser) extends Page with Browser {
   val url = s"$baseUrl/join/supporter/enter-details"
 
-  def pageHasLoaded: Boolean = pageHasElement(id("cc-cvc"))
+  def pageHasLoaded: Boolean = pageHasElement(className("js-sign-in-note"))
 
   def userIsSignedIn: Boolean = elementHasText(userDisplayName, testUser.username)
 
   def fillInDeliveryAddress() { DeliveryAddress.fillIn() }
 
-  def fillInCardDetails() { CreditCard.fillIn() }
+  def clickContinue() = clickOn(className("js-continue-name-address"))
 
-  def fillInCardDeclined() { CreditCard.fillInCardDeclined() }
+  def fillInCreditCardPaymentDetailsStripe(): Unit = StripeCheckout.fillIn()
 
-  def fillInCardDeclinedFraud() { CreditCard.fillInCardDeclinedFraud() }
+  def changeCountry(country: String) = { DeliveryAddress.selectCountryCode(country) }
 
-  def fillInCardDeclinedCvc() { CreditCard.fillInCardDeclinedCvc() }
+  def currencyHasChanged(): Boolean = hiddenElementHasText(currency, "US$")
 
-  def fillInCardDeclinedExpired() { CreditCard.fillInCardDeclinedExpired() }
+  def pay() = clickOn(className("js-stripe-checkout"))
 
-  def fillInCardDeclinedProcessError() { CreditCard.fillInCardDeclinedProcessError() }
+  def switchToStripe() = driver.switchTo().frame(driver.findElement(StripeCheckout.container.by))
```

Comment:
Please do not use `driver.findElement` directly. Create another public method in `acceptance.util.Browser` trait for switching frames and use that:

```
  def switchFrame(q: Query) {
    if (pageHasElement(q))
      driver.switchTo().frame(q.webElement)
    else
      throw new MissingPageElementException(q)
  }
```